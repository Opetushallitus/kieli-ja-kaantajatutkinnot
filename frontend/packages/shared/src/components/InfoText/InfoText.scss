.info-text {
  & &__icon {
    margin-right: 0.5rem;
  }

  & &__label {
    font-size: 1.3rem;
    margin-top: 2px;
  }
}
