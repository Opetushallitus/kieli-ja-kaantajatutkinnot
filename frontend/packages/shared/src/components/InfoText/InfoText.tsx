import { InfoOutlined as InfoIcon } from '@mui/icons-material';
import { FC, PropsWithChildren } from 'react';

import { Color } from '../../enums/common';
import { Text } from '../Text/Text';
import './InfoText.scss';

export type InfoTextProps = PropsWithChildren;

export const InfoText: FC<InfoTextProps> = ({ children }) => {
  return (
    <div className="info-text columns">
      <InfoIcon className="info-text__icon" color={Color.Secondary} />
      <Text className="info-text__label">{children}</Text>
    </div>
  );
};
