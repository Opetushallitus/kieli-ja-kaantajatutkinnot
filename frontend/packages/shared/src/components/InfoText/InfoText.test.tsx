import renderer from 'react-test-renderer';

import { InfoText } from './InfoText';

describe('InfoText', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<InfoText>This is an info text</InfoText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
