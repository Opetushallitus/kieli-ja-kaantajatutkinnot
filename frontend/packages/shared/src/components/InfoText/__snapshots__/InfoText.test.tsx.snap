// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoText should render correctly 1`] = `
<div
  className="info-text columns"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeMedium info-text__icon css-kw0y8f-MuiSvgIcon-root"
    data-testid="InfoOutlinedIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M11 7h2v2h-2zm0 4h2v6h-2zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8"
    />
  </svg>
  <p
    className="MuiTypography-root MuiTypography-body1 info-text__label css-ahj2mt-MuiTypography-root"
  >
    This is an info text
  </p>
</div>
`;
